File: GameOfLife.CSharp/Renderer.cs
using System;
using System.Text;

namespace GameOfLife.CSharp
{
    public class Renderer
    {
        public void Render(Board board, int generation, TimeSpan renderTime)
        {
            Console.Clear();
            var sb = new StringBuilder();

            for (int y = 0; y < board.Height; y++)
            {
                for (int x = 0; x < board.Width; x++)
                    Render(sb, board[x, y]);

                sb.AppendLine();
            }

            Console.WriteLine($"Generation: {generation}, Render Time: {renderTime.TotalMilliseconds}ms");
            Console.WriteLine(sb);
        }

        void Render(StringBuilder sb, CellState cell)
        {
            char c = cell == CellState.Alive ? 'x' : ' ';
            sb.Append(c);
        }
    }
}

File: GameOfLife.CppOpenMP/Renderer.cpp
#include "Renderer.h"
#include <cstdlib>
#include <string>
#include <sstream>
#include <iostream>

void Renderer::Render(Board & board, int generation, int time)
{
	system("cls");
	std::stringstream ss;

	ss << "Generation: " << generation << ", Render Time: " << time << "ns" << std::endl;


	for (int y = 0; y < board.GetHeight(); y++)
	{
		for (int x = 0; x < board.GetWidth(); x++)
			ss << (board.Get(x, y) == Alive ? 'x' : ' ');
		ss << std::endl;
	}

	std::cout << ss.str();
}
